packit_service/worker/jobs.py

```diff
@@ -473,10 +474,17 @@ def is_project_public_or_enabled_private(self) -> bool:
 
     def get_jobs_matching_event(self) -> List[JobConfig]:
         """
-        Get list of non-duplicated all jobs that matches with event's trigger.
+        Get list of non-duplicated all jobs that matches with event's trigger and identifier.
         """
         jobs_matching_trigger = []
         for job in self.event.package_config.jobs:
+            # FIXME: perhaps this should be done more "abstractly"?
+            #        same for CheckRerunEvent in get_handlers_for_event
+            if (
+                isinstance(self.event, TestingFarmResultsEvent)
+                and self.event.identifier != job.identifier
```

Comment:
would it make sense to run this in the `pre_check` method of the `TestingFarmResultsHandler`? otherwise, LGTM

Reply:
So testing turned out to be more complicated than I expected due to where `pre_check` is called -- it is necessary to test `create_tasks` which uses a lot of utility functions (dumping package config, etc) + celery stuff. I tried to remove most of this through mocks and focus on the number of tasks created. Could someone please re-review the changes?